[Devansh Shah] iP

Devanshshah1309 opened the pull request:
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Devanshshah1309 committed:
Add Greet, Echo, Exit

Completed Level 1

Devanshshah1309 committed:
Add, List

Completed Level 2

Devanshshah1309 committed:
Level 3

Mark as Done

Devanshshah1309 committed:
Fix task count bug

Devanshshah1309 committed:
Add Deadline Class

Devanshshah1309 committed:
Add Event Class

Devanshshah1309 committed:
Add ToDo class

Devanshshah1309 committed:
Refactor Task Class to Accommodate Children

Devanshshah1309 committed:
Complete Level 4

Refactored and Improved Regex pattern matching

Devanshshah1309 committed:
Update .gitignore

Devanshshah1309 committed:
Remove Duke Logo

Also, shift Scanner outside while loop

Devanshshah1309 committed:
Improve grammar in Task

Devanshshah1309 committed:
Semi-automate testing

Added EXPECTED.TXT and input.txt

Devanshshah1309 committed:
Add Basic Handling of Errors

Introduce DukeException

Devanshshah1309 committed:
Add delete functionality

Devanshshah1309 committed:
Use Enums

Devanshshah1309 committed:
Refactor catching DukeException

Devanshshah1309 committed:
Close scanner in finally block

Devanshshah1309 committed:
Refactor Duke into switch statements

Remove regex and improve error handling

Devanshshah1309 committed:
Follow style guide and coding conventions

Devanshshah1309 committed:
Improve OOP Design of Duke

Also improved exception handling and formatting of output

Devanshshah1309 committed:
Save tasks to file

Created a FileOperations class to perform file manipulation tasks.

Saving the tasks allows persistence of data even after the program stops running

Devanshshah1309 committed:
Merge branch 'branch-Level-7' into master

Devanshshah1309 committed:
Store deadlines and timings as LocalDateTime

Devanshshah1309 committed:
Misc fixes

Devanshshah1309 committed:
Merge branch 'branch-Level-8' into master

Devanshshah1309 committed:
Resolve merge conflicts

Allow storing dates in data.txt to convert back and forth between data dnd string representations.

Devanshshah1309 committed:
Resolve merge conflicts

Devanshshah1309 committed:
Create TaskList class

The TaskList class handles addition, deletion of tasks, getting the count, and listing of all tasks

Devanshshah1309 committed:
Refactor FileOperations to Storage class

Devanshshah1309 committed:
Stop tracking data.txt

Devanshshah1309 committed:
Abstract parse operations into Parser

Devanshshah1309 committed:
Abstract UI operations

Devanshshah1309 committed:
Organize classes into duke package

Devanshshah1309 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Devanshshah1309 committed:
Automate project build using Gradle

Devanshshah1309 committed:
Add JUnit tests for Todo and Parser

Devanshshah1309 committed:
Add JavaDoc comments to most methods and classes

Devanshshah1309 committed:
Follow Coding Standard

Devanshshah1309 committed:
Merge branch 'branch-A-JavaDoc' into master

Devanshshah1309 committed:
Merge branch 'branch-A-CodingStandard' into master

Devanshshah1309 committed:
Add find functionality

Devanshshah1309 committed:
Merge branch 'branch-Level-9' into master

Devanshshah1309 committed:
Add Checkstyle using gradle

Devanshshah1309 committed:
Merge branch 'branch-A-CheckStyle' into master

Devanshshah1309 committed:
Add dependency for javafx

Devanshshah1309 committed:
Add basic UI for Duke

Devanshshah1309 committed:
Add user image and improve UI

Devanshshah1309 committed:
Add image of duke mascot

Devanshshah1309 committed:
Correct naming of image

Devanshshah1309 committed:
Use FXML for GUI

Devanshshah1309 committed:
Merge branch 'branch-Level-10' into master

Devanshshah1309 committed:
Improve UI/UX

Added color and fixed minor UI bugs.

Tan-Jin-Waye commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,296 @@
+
+        try {
+            file = new File(FILE_NAME);
+            if (file.createNewFile()) {
```
Not sure if this is a coding standard violation, though if it is I can propose one (obvious) change:

~~~java

boolean fileIsValid = file.createNewFile();
if (fileIsValid) {
..

~~~

Edit: noticed several other instances using boolean return from functions  in if/while statements. Not sure if they will be that pedantic but I suppose apply above changes similarly.

Tan-Jin-Waye commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,296 @@
+////        // You will need to import `javafx.scene.layout.Region` for this.
+////        dialogContainer.setPrefHeight(Region.USE_COMPUTED_SIZE);
+////
+////        userInput.setPrefWidth(325.0);
+////
+////        sendButton.setPrefWidth(55.0);
+////
+////        AnchorPane.setTopAnchor(scrollPane, 1.0);
+////
+////        AnchorPane.setBottomAnchor(sendButton, 1.0);
+////        AnchorPane.setRightAnchor(sendButton, 1.0);
+////
+////        AnchorPane.setLeftAnchor(userInput, 1.0);
+////        AnchorPane.setBottomAnchor(userInput, 1.0);
+////
+////        sendButton.setOnMouseClicked((event) -> {
+////            dialogContainer.getChildren().add(getDialogLabel(userInput.getText()));
+////            userInput.clear();
+////        });
+////
+////        userInput.setOnAction((event) -> {
+////            dialogContainer.getChildren().add(getDialogLabel(userInput.getText()));
+////            userInput.clear();
+////        });
+//
+//        //Part 3. Add functionality to handle user input.
+////        sendButton.setOnMouseClicked((event) -> {
+////            handleUserInput();
+////        });
+////
+////        userInput.setOnAction((event) -> {
+////            handleUserInput();
+////        });
+////
+////        dialogContainer.heightProperty().addListener((observable) -> scrollPane.setVvalue(1.0));
+////
+////        scene = new Scene(mainLayout);
+////        stage.setScene(scene);
+////        stage.show();
+//    }
```
Consider removing this function as soon as you're certain it won't be needed anymore, as well as other occurrences in code.

Tan-Jin-Waye commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,108 @@
+        }
+    }
+
+    public int count() {
```
Consider renaming to something like getListSize/getSize in line with getters.

Tan-Jin-Waye commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,108 @@
+        return tasks.size();
+    }
+
+    public Task get(int taskNumber) {
```
Consider renaming to something like getTaskAtIndex/getTask in line with getters.

Tan-Jin-Waye commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,25 @@
-| |_| | |_| |   <  __/
-|____/ \__,_|_|\_\___|
-
+	-------------------------------
```
Consider adding this file to .gitignore.

Tan-Jin-Waye reviewed (commented):
Strong adherence to coding standards. Really had to work to surface even minor nitpicks. Keep it up and ATB! 👍

Devanshshah1309 committed:
Improve code quality

Got rid of dead code to improve readability and adhere to coding standards

Devanshshah1309 committed:
Add assertion

Ensure that the description of a task is never empty because that would be meaningless

Devanshshah1309 committed:
Merge pull request #2 from Devanshshah1309/branch-A-Assertions

Assert that task description is non-empty

Devanshshah1309 committed:
Remove arrowhead nesting

Devanshshah1309 committed:
Merge remote-tracking branch 'origin' into branch-A-CodeQuality

Devanshshah1309 committed:
Merge pull request #1 from Devanshshah1309/branch-A-CodeQuality

Improve code quality by getting rid of dead code and also refactoring if-else statements to remove arrowheads.

Devanshshah1309 committed:
Add functionality to detect duplicate tasks

Devanshshah1309 committed:
Add delay functionality before exiting

Devanshshah1309 committed:
Improve code quality

Devanshshah1309 committed:
Add User guide and Ui.png

Devanshshah1309 committed:
Fix minor gradle issue
